hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java

```diff
@@ -109,6 +113,10 @@ public SqlResult execute(@Nonnull SqlStatement statement) {
         }
     }
 
+    private boolean skipUpdateStatistics() {
+        return connectionType.equals(ConnectionType.MC_JAVA_CLIENT);
```

Comment:
Minor: Maybe we can store the result of this comparison instead of the `connectionType`, in a field called `skipUpdateStatistics`. That way, we can eliminate the need to perform this comparison in each execute operation